packages/utils/perf/crawl-file-system/index.ts

```diff
@@ -6,17 +6,18 @@ import {
 } from '../../src/lib/file-system';
 import { crawlFileSystemOptimized0 } from './optimized0';
 
+// TODO pls double check me here... also, is pop that much faster that ESLint should be disabled instead?
 const PROCESS_ARGUMENT_TARGET_DIRECTORY = String(
   process.argv
     .find(arg => arg.startsWith('--directory'))
     ?.split('=')
-    .pop() || '',
+    .slice(0, -1)[0] ?? '',
 );
 const PROCESS_ARGUMENT_PATTERN = String(
   process.argv
     .find(arg => arg.startsWith('--pattern'))
     ?.split('=')
-    .pop() || '',
+    .slice(0, -1)[0] ?? '',
 );
```

Comment:
The problem is `pop` mutates an array. However, here it doesn't really matter, because it's an intermediate array created by `split`, which is immediately mutated and never used again (can't be, since it's not assigned to a variable). Unfortunately, the ESLint rule isn't always able to distinguish these cases.

I think there is a better solution than either `.pop()` or `.slice(0, -1)[0]` - use `.at(-1)` to get last array item :slightly_smiling_face: 

(Oh and on an unrelated note - wrapping the whole thing in `String()` is almost certainly pointless. The underlying expression cannot return anything other than a string.)